librz/core/cmd.c

```diff
@@ -4760,55 +4761,123 @@ DEFINE_HANDLE_TS_FCN_AND_SYMBOL(redirect_command) {
 	return res;
 }
 
+typedef struct _search_help {
+	bool color;
+	RzStrBuf *sb;
+	PJ *pj;
+} RzHelpSearch;
+
+static bool help_search_cmd_desc_entry(RzCmd *cmd, const RzCmdDesc *cd, void *user) {
+	rz_return_val_if_fail(cd, false);
+	if (cd->type != RZ_CMD_DESC_TYPE_ARGV &&
```

Comment:
This whole check can be easily removed and everything still works. It has the advantage that you also get the summary of oldinput commands, though you can't ofc descend into them.

Reply:
However, I would add a a "type" field in the json output to export the fact that some commands are OLDINPUT, others are ARGV and others ARGV_MODES.

Reply:
i would not do this. the only reason i'm doing something like this, is to not handle OLDINPUT and the fake one.

Reply:
Fake ones cannot arrive at this point because the foreach method discards them.
For oldinput ones, you don't need to handle anything in particular and it will provide slightly better results.

By removing this check:
```
[0x00000000]> w?*
| w <string>           # Write string
| wB <value>           # Set bits with given value
| wB- <value>          # Unset bits with given value
| wv <value>           # Write value as 4-bytes/8-bytes based on value
| wv1 <value>          # Write value of 1 byte
| wv2 <value>          # Write value of 2 byte
| wv4 <value>          # Write value of 4 byte
| wv8 <value>          # Write value of 8 byte
| w0 <len>             # Write <len> bytes with value 0x00
| w1+ [<n>]            # Increment a byte
| w1- [<n>]            # Decrement a byte
| w2+ [<n>]            # Increment a word
| w2- [<n>]            # Decrement a word
| w4+ [<n>]            # Increment a dword
| w4- [<n>]            # Decrement a dword
| w8+ [<n>]            # Increment a qword
| w8- [<n>]            # Decrement a qword
| w6d <base64>         # Write the base64-decoded bytes
| w6e <hexstring>      # Write the base64-encoded bytes
| wh                # whereis/which shell command
| we                # Extend write operations (insert bytes instead of replacing)
| wu                # Apply unified hex patch (see output of cu)
| wr                # Write <num> random bytes
| wA                # Alter/modify opcode at current seek (see wA?)
| wc                # Write cache commands
| wz                # Write zero-terminated string
| wt                # Write to file
| wf                # Write data from file, socket, offset
| ww                # Write wide string
| wx                # Write hexadecimal
| wa                # Write opcode, separated by ';'
| wb                # Write in current block with cyclic hexstring
| wm                # Set binary mask hexpair to be used as cyclic write mask
| wo                # Write in block with operation
| wd                # Duplicate N bytes from offset at current seek
| ws                # Write 1 byte for length and then the string
```
With this check instead:
```
| w <string>           # Write string
| wB <value>           # Set bits with given value
| wB- <value>          # Unset bits with given value
| wv <value>           # Write value as 4-bytes/8-bytes based on value
| wv1 <value>          # Write value of 1 byte
| wv2 <value>          # Write value of 2 byte
| wv4 <value>          # Write value of 4 byte
| wv8 <value>          # Write value of 8 byte
| w0 <len>             # Write <len> bytes with value 0x00
| w1+ [<n>]            # Increment a byte
| w1- [<n>]            # Decrement a byte
| w2+ [<n>]            # Increment a word
| w2- [<n>]            # Decrement a word
| w4+ [<n>]            # Increment a dword
| w4- [<n>]            # Decrement a dword
| w8+ [<n>]            # Increment a qword
| w8- [<n>]            # Decrement a qword
| w6d <base64>         # Write the base64-decoded bytes
| w6e <hexstring>      # Write the base64-encoded bytes

```